tools/golden_tests_harvester/test/golden_tests_harvester_test.dart

```diff
@@ -2,4 +2,147 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-void main() {}
+import 'dart:async';
+import 'dart:io' as io;
+
+import 'package:golden_tests_harvester/golden_tests_harvester.dart';
+import 'package:litetest/litetest.dart';
+import 'package:path/path.dart' as p;
+
+void main() async {
+  Future<void> withTempDirectory(FutureOr<void> Function(io.Directory) callback) async {
+    final io.Directory tempDirectory = await io.Directory.systemTemp.createTemp('golden_tests_harvester_test.');
+    try {
+      await callback(tempDirectory);
+    } finally {
+      await tempDirectory.delete(recursive: true);
+    }
+  }
+
+  test('should require a file named "digests.json" in the working directory', () async {
+    await withTempDirectory((io.Directory tempDirectory) async {
+      final StringSink stderr = StringBuffer();
+      try {
```

Comment:
You use this pattern a few times.  You could factor this out to 
```dart
expectThrow<io.FileSystemException>((){ await harvest(...);}, (io.FileExceptionException e) {
  expect(e.path, contains('digest.json'));
});
```

Reply:
Reasonable suggestion, done!